account_move_line_accounting_description_sale/models/sale_order.py

```diff
@@ -0,0 +1,13 @@
+from odoo import models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    def _prepare_invoice_line(self, **optional_values):
+        line = super(SaleOrderLine, self)._prepare_invoice_line(**optional_values)
+        line["external_name"] = line["name"]
+        product = self.env["product.product"].search([("id", "=", line["product_id"])])
```

Comment:
```suggestion
        product_id = line.get("product_id")
        if product_id:
            product = self.env["product.product"].browse(line["product_id"])
```
`browse` is better than `search` if you already have the id.
Also if the line is a note or a section `line["product_id"]` might throw a `KeyError` here.